lighthouse-core/gather/gather-runner.js

```diff
@@ -167,6 +167,34 @@ class GatherRunner {
     }
   }
 
+  /**
+   * Returns an error if we ended up on the `chrome-error` page and all other requests failed.
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @return {LH.LighthouseError|undefined}
+   */
+  static getInterstitialError(networkRecords) {
+    const interstitialRequest = networkRecords
+      .find(record => record.documentURL.startsWith('chrome-error://'));
+    // If the page didn't end up on a chrome interstitial, there's no error here.
+    if (!interstitialRequest) return undefined;
+
+    const pageNetworkRecords = networkRecords
+      .filter(record => !URL.NON_NETWORK_PROTOCOLS.includes(record.protocol) &&
+        !record.documentURL.startsWith('chrome-error://'));
+    // If at least one request didn't fail, there's no error here.
```

Comment:
this seems...weird? Is this catching like having an iframe with a bad cert but the page loaded fine or some other case? Maybe it would help to expand the comment to mention what this is expecting

Reply:
I'm not aware of any situation where this is possible, just seemed like a good safety net if they started on the error interstitial or something

Reply:
added a comment to that effect

Reply:
I guess maybe the comment is confusing me?

> If at least one request didn't fail, there's no error here.

but

```js
if (!pageNetworkRecords.some(record => record.failed)) return undefined;
```

means that if *all* of the requests didn't fail there's no error here?

Reply:
oh hahah yes very poorly written I was trying to say "If we didn't see at least one request fail, there's no error here"

Reply:
```suggestion
    // If none of the requests failed, there's no error here.
```

Reply:
ah, ok, somehow the comment made me unable to read the code. Your nonchalance about the redirect case makes so much more sense now since it is handled :)